include/libtorrent/session.hpp

```diff
@@ -149,6 +149,8 @@ namespace libtorrent
 		libtorrent::dht_settings dht_settings;
 
 		dht::dht_storage_constructor_type dht_storage_constructor;
+
+		bdecode_node state;
```

Comment:
probably not in a bencoded ``entry`` object. I think something more agnostic to how it's currently stored in bencoded form. i.e. a vector of nodes and a ``node_id`` for the last session's id. etc.